### DISCUSSION TITLE
Qfield and actions on a layer

### DISCUSSION THREAD

**Discussion by geguill** [opengisch/QField#5849] [Q&A]
Hello
is QField able to execute actions on a layer, for example, open a URL?
If yes, how do we do it?
Thanks.
Best regards

**Comment by VxTedxV** [comment]
You can set a form where you put a link on or a field showing a link for each feature.

**Comment by geguill** [comment]
[essai.zip](https://github.com/user-attachments/files/17934831/essai.zip)
Thanks for the answer, but I'm still stuck.
I created a project in QGIS named "essai" (see attachment), based on the layer "points_google", on which I created 2 actions.
These 2 actions appear in the form.
When I open the project in QField (after packaging it with QField Sync or not), I don't find the 2 actions.
Is this normal?
Thanks

**Comment by geguill** [comment]
hello,
could you give me an example of a project with a layer on which an action is implemented, which can be found on QField?
Thanks

**Comment by VxTedxV** [comment]
I think you can find an answer here : #4615

**Comment by nirvn** [comment]
Clarification: QField does not yet support vector layer actions as configured within a vector layer properties dialog's actions panel. And while a few action types will most likely never been implemented (windows/macos/linux command for e.g.), I can see the open URL ones as something that'd be worth implementing one day.

What @VxTedxV was suggesting above is for you to not rely on vector layer actions but instead a text or QML widget within which you'd have hyperlinks or buttons to mimic actions.

**Comment by geguill** [comment]
Thank you for this valuable information.
Kind regards

**Comment by geguill** [comment]
Thanks for your answer.
best regards

**Reply by nirvn** [reply]
@geguill , were you able to achieve what you aimed at doing?

**Comment by geguill** [comment]
Yes,
thank you.
I created a QML widget that QField runs without problems

**Reply by nirvn** [reply]
@geguill , glad to hear this. Happy field mapping!

@VxTedxV , thanks for helping others in this growing QField community!
